client/history/client.go

```diff
@@ -846,24 +856,37 @@ func (c *clientImpl) GetReplicationMessages(
 				}
 				return
 			}
-			respChan <- resp
-		}(requestContext, client, req)
+			_, responseInfo := rpc.ContextWithResponseInfo(requestContext)
```

Comment:
I'm afraid this will not work. You need to use context returned from `ContextWithResponseInfo` when making an rpc call, otherwise the responseInfo structure will remain empty.

Reply:
Just realize this use as a callback. Update d the code.